### ISSUE TITLE
Issue rendering tables with leading paragraph

### ISSUE THREAD

**Issue by olipo186** [lunet-io/markdig#370]
Hi,

I noticed this different while rendering some markdown markup into HTML. Is this expected behavior? Shouldn't tables be rendered properly regardless of whether there's text just before it or not?

I'm converting using the following;
```
var pipeline = new MarkdownPipelineBuilder().UseAdvancedExtensions().Build();
var html = Markdown.ToHtml(markdown, pipeline);
```

**Test case 1: No extra line break added**

Input
```
Hello world
| a | b |
|---|---|
| c | d |
```

Output
```
<p>Hello world | a | b | |---|---| | c | d |</p>
```

**Test case 2: Extra line break added**

Input
```
Hello world

| a | b |
|---|---|
| c | d |
```

Output
```
<p>Hello world</p>
<table>
    <thead>
        <tr>
            <th>a</th>
            <th>b</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>c</td>
            <td>d</td>
        </tr>
    </tbody>
</table>
```

**Comment by olipo186** [comment]
I agree with your conclusion. This seems to be an edge case that is handled slightly different by different markdown parsers. Most of the parsers seem to align with the behavior of Markdig however, and I haven't been able to find any "spec" for the markdown table syntax that covers this.

Thanks for your reply @xoofx, feel free to close the issue!

**Comment by edwardcookemacu** [comment]
Now that github renders the tables with a leading line with text as expected, a table underneath it, is this something worth looking into?

For example
```markdown
sometext
|col1|col2|
|-|-|
|a|line|
```
## Markdig rendered output:
```html
<p>sometext
|col1|col2|
|-|-|
|a|line|</p>
```

## Github:

sometext
|col1|col2|
|-|-|
|a|line|
